simtools/model/telescope_model.py

```diff
@@ -211,49 +213,11 @@ def _setConfigFileDirectory(self):
     def _loadParametersFromDB(self):
         ''' Read parameters from DB and store them in _parameters. '''
 
-        def _readParsFromOneType(telescopeType):
-            '''
-            Read parameters from DB for one specific type (telescopeTYpe, site ...).
-
-            Parameters
-            ----------
-            telescopeType: str
-
-            Returns
-            -------
-            dict containing the parameters
-            '''
-            fileNameDB = 'parValues-{}.yml'.format(telescopeType)
-            yamlFile = cfg.findFile(fileNameDB, self._modelFilesLocations)
-            logger.debug('Reading DB file {}'.format(yamlFile))
-            with open(yamlFile, 'r') as stream:
-                pars = yaml.load(stream, Loader=yaml.FullLoader)
-            return pars
-
-        def _collectAplicablePars(pars):
-            '''
-            Collect only Applicable parameters from pars and store them into _parameters
-
-            Parameters
-            ----------
-            pars: dict
-                Dict with the whole list of parameters of a certain telescope type.
-            '''
-            for parNameIn, parInfo in pars.items():
-                if parInfo['Applicable']:
-                    parNameOut, parValueOut = self._validateParameter(
-                        parNameIn,
-                        parInfo[self.version]
-                    )
-                    self._parameters[parNameOut] = parValueOut
-
-        parametersDB = _readParsFromOneType(telescopeType=self.telescopeType)
-        _collectAplicablePars(parametersDB)
-
-        if getTelescopeSize(self.telescopeType) == 'MST':
-            logger.debug('Telescope is MST type - reading optics parameters')
-            parametersDB = _readParsFromOneType(telescopeType='MST-optics')
-            _collectAplicablePars(parametersDB)
+        self._parameters = db.getModelParameters(
```

Comment:
I wonder about this a bit when we start using the DB. Maybe it will be more efficient to read the DB once for all telescopes instead of reading each telescope separately. For now we can leave it like this, but it would be good to discuss it and then run some benchmarking to make sure.

Reply:
Anyway, what the telescope model needs here are the parameters from one telescope. You may want to split the getModelParameters into two functions in db_handler, one that returns all the parameters and another one that select the ones for a given telescope type.

Reply:
Yeah, this was just me wondering where it should go. We'll see later, perhaps in the db_handler it's OK.